src/mapping/parser/mod.rs

```diff
@@ -385,19 +373,20 @@ fn query_from_pair(pair: Pair<Rule>) -> Result<Box<dyn query::Function>> {
             inner_quoted_string_escaped_from_pair(pair.into_inner().next().ok_or(TOKEN_ERR)?)?,
         ))),
         Rule::null => Box::new(Literal::from(Value::Null)),
-        Rule::number => Box::new(Literal::from(Value::from(
-            pair.as_str().parse::<f64>().unwrap(),
-        ))),
+        Rule::number => Box::new(Literal::from(
+            pair.as_str()
+                .parse::<i64>()
+                .map(Value::from)
+                .unwrap_or_else(|_| Value::from(pair.as_str().parse::<f64>().unwrap())),
+        )),
```

Comment:
We now try to parse a number to an integer, and only if that fails do we parse to a float. The float always works, as the parser prevents invalid numbers from generating the `Rule::number` rule.

Reply:
Semantically that does seem like the more "correct" approach. I'll make the change.

Reply:
resolved in ec1f9b8